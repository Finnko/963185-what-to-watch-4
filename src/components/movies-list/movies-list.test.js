import React from 'react';
import renderer from 'react-test-renderer';
import {createMemoryHistory} from 'history';
import {Router} from 'react-router-dom';
import MoviesList from './movies-list';

const mock = [
  {
    id: 1,
    title: `The Grand Budapest Hotel`,
    genre: `Drama`,
    thumb: `img/bohemian-rhapsody.jpg`,
    releaseYear: 2014,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    backgroundImage: `img/bg-the-grand-budapest-hotel.jpg`,
    description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
    director: `Michael Bay`,
    actors: [`Leonardo Di Caprio`],
    rating: 7.5,
    ratingCount: 250,
    poster: `/img/the-grand-budapest-hotel-poster.jpg`,
  }
];

describe(`MoviesList component render correctly`, () => {
  it(`Should MoviesList component render correctly`, () => {
    const history = createMemoryHistory();

    const tree = renderer
      .create(
          <Router history={history}>
            <MoviesList
              movies={mock}
              viewMode={`WITH_PLAYER`}
            />
          </Router>, {
            createNodeMock: () => {
              return {};
            },
          }
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
