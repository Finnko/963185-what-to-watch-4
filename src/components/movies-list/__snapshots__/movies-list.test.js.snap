// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesList component render correctly Should MoviesList component render correctly 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <a
      className="small-movie-card__link"
      href="/movie/1"
      onClick={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <video
          height="175"
          muted={true}
          poster="img/bohemian-rhapsody.jpg"
          src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
          width="100%"
        />
      </div>
    </a>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="/movie/1"
        onClick={[Function]}
      >
        The Grand Budapest Hotel
      </a>
    </h3>
  </article>
</div>
`;
